### PR TITLE
Feature/spinners

### PR DESCRIPTION
**Pull request by ElishaSamPeterPrabhu**
#36

### CODE REVIEW

**Comment by BAL4-R** [comment]
@ElishaSamPeterPrabhu could you link the ticket in the description?
Use "closes #[ticket-number]" so that the ticket will be closed automatically when the pr is merged. ([closing keywords](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword))

File: DemoApp/DemoApp/Views/TMSpinnerPage.xaml [hunk 1]
```diff
@@ -0,0 +1,31 @@
+<?xml version="1.0" encoding="utf-8" ?>
+<ContentPage xmlns="http://schemas.microsoft.com/dotnet/2021/maui"
+             xmlns:x="http://schemas.microsoft.com/winfx/2009/xaml"
+             x:Class="DemoApp.Views.TMSpinnerPage"
+             xmlns:modus="clr-namespace:Trimble.Modus.Components;assembly=Trimble.Modus.Components"
+             Title="TMSpinnerPage">
+  <VerticalStackLayout BackgroundColor="Tan" Spacing="50" Padding="10">
+    <Label Text="Determinate Spinner"></Label>
```
Comment (Sofiya-kumar) [inline DemoApp/DemoApp/Views/TMSpinnerPage.xaml:8]: Code formatting is required.

**Review by Sofiya-kumar — CHANGES_REQUESTED** [review]

**Comment by ElishaSamPeterPrabhu** [comment]
> @ElishaSamPeterPrabhu could you link the ticket in the description? Use "closes #[ticket-number]" so that the ticket will be closed automatically when the pr is merged. ([closing keywords](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword))

Done

File: DemoApp/DemoApp/DemoApp.csproj [hunk 2]
```diff
@@ -112,6 +112,9 @@
 	  <MauiXaml Update="Views\TMRadioButtonPage.xaml">
```
Comment (rthanga1) [inline DemoApp/DemoApp/DemoApp.csproj:112]: Not working in iOS. OnPaintSurface never gets called when running in iOS.

Simulator Screen Shot - iPhone 14 - 2023-06-27 at 15 41 28

File: Trimble.Modus.Components/Controls/TMSpinner/TMSpinner.xaml.cs [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+        #region Private Methods
+        private static void OnSpinnerColorChanged(BindableObject bindable, object oldValue, object newValue)
+        {
+            if (bindable is TMSpinner tmSpinner)
+            {
+                tmSpinner._spinnerColor = ((SpinnerColor)newValue == SpinnerColor.White) ? ResourcesDictionary.ColorsDictionary(ColorsConstants.White)
+                    : ResourcesDictionary.ColorsDictionary(ColorsConstants.TrimbleBlue);
+            }
+        }
+        private static void OnSpinnerChanged(BindableObject bindable, object oldValue, object newValue)
```
Comment (rthanga1) [inline Trimble.Modus.Components/Controls/TMSpinner/TMSpinner.xaml.cs:64]: Should be `OnSpinnerTypeChanged `

File: Trimble.Modus.Components/Controls/TMSpinner/TMSpinner.xaml.cs [hunk 4]
```diff
@@ -0,0 +1,143 @@
...
+            {
+                _startAngle += 5f;
+                _sweepAngle += 5f;
+                if (_sweepAngle >= 360f)
+                {
+                    _sweepAngle = 0f;
+                    _startAngle = 0f;
+                }
+            }
+            Device.BeginInvokeOnMainThread(() =>
```
Comment (rthanga1) [inline Trimble.Modus.Components/Controls/TMSpinner/TMSpinner.xaml.cs:92]: I don't think we need the `Device.BeginInvokeOnMainThread( ` here, please check once. Even if we need, Device.BeginInvokeOnMainThread is deprecated now, we shouldn't use it.

File: Trimble.Modus.Components/Controls/TMSpinner/TMSpinner.xaml.cs [hunk 5]
```diff
@@ -0,0 +1,143 @@
...
+            }
+            else if (_spinnerType == SpinnerType.Determinate)
+            {
+                float startAngle = 270f;
+                float progress = 1f;
+                float endAngle = startAngle + (_sweepAngle * progress);
+                canvas.DrawArc(rect, startAngle, endAngle - startAngle, false, arcPaint);
+            }
+        }
+        #endregion
```
Comment (rthanga1) [inline Trimble.Modus.Components/Controls/TMSpinner/TMSpinner.xaml.cs:141]: Clear the `canvas` after drawn. Check if a clear or flush call available

File: Trimble.Modus.Components/Enums/SpinnerColor.cs [hunk 6]
```diff
@@ -0,0 +1,6 @@
+﻿namespace Trimble.Modus.Components.Enums;
+public enum SpinnerColor
+{
+    White,
```
Comment (rthanga1) [inline Trimble.Modus.Components/Enums/SpinnerColor.cs:4]: Should be `Primary` and `Secondary`

**Review by rthanga1 — CHANGES_REQUESTED** [review]

File: DemoApp/DemoApp/Views/TMSpinnerPage.xaml (hunk 1 above)
Comment (ElishaSamPeterPrabhu) [reply DemoApp/DemoApp/Views/TMSpinnerPage.xaml]: Done

File: Trimble.Modus.Components/Controls/TMSpinner/TMSpinner.xaml.cs (hunk 3 above)
Comment (ElishaSamPeterPrabhu) [reply Trimble.Modus.Components/Controls/TMSpinner/TMSpinner.xaml.cs:64]: Renamed

File: Trimble.Modus.Components/Controls/TMSpinner/TMSpinner.xaml.cs (hunk 4 above)
Comment (ElishaSamPeterPrabhu) [reply Trimble.Modus.Components/Controls/TMSpinner/TMSpinner.xaml.cs:92]: We need it hence changed to MainThread.BeginInvokeOnMainThread

File: Trimble.Modus.Components/Controls/TMSpinner/TMSpinner.xaml.cs (hunk 5 above)
Comment (ElishaSamPeterPrabhu) [reply Trimble.Modus.Components/Controls/TMSpinner/TMSpinner.xaml.cs:141]: Using canvas.flush()

File: Trimble.Modus.Components/Enums/SpinnerColor.cs (hunk 6 above)
Comment (ElishaSamPeterPrabhu) [reply Trimble.Modus.Components/Enums/SpinnerColor.cs:4]: Renamed to Primary(Blue),Secondary(White)

File: DemoApp/DemoApp/DemoApp.csproj (hunk 2 above)
Comment (rthanga1) [reply DemoApp/DemoApp/DemoApp.csproj:112]: I changed the linking behaviour and it worked.

**Review by rthanga1 — APPROVED** [review]

**Pull request merged**